flow/src/test/java/com/sequenceiq/flow/service/flowlog/FlowLogDBServiceTest.java

```diff
@@ -248,32 +261,72 @@ public void doNotCancelTooYoungTerminationFlowForResourceTest() {
+        assertThat(savedFlowLog.getFinalized()).isTrue();
+        assertThat(savedFlowLog.getStateStatus()).isEqualTo(StateStatus.SUCCESSFUL);
+        assertThat(savedFlowLog.getOperationType()).isEqualTo(OperationType.DIAGNOSTICS);
+        assertThat(savedFlowLog.getCloudbreakNodeId()).isEqualTo(NODE_ID);
+        assertThat(savedFlowLog.getVariables()).isNull();
+        assertThat(savedFlowLog.getVariablesJackson()).isNull();
+    }
+
+    @Test
+    void testClose() throws TransactionExecutionException {
+        prepareFinalization();
+        Map<Object, Object> params = Map.of("param1", StateStatus.SUCCESSFUL, "param2", 234L, "param3", "true");
+        underTest.close(ID, FLOW_ID, false, params);
+
+        verifyFinalization();
+        FlowLog savedFlowLog = savedFlowLogCaptor.getValue();
+        assertThat(savedFlowLog.getResourceId()).isEqualTo(ID);
+        assertThat(savedFlowLog.getFlowId()).isEqualTo(FLOW_ID);
+        assertThat(savedFlowLog.getCurrentState()).isEqualTo(FINISHED_STATE);
+        assertThat(savedFlowLog.getFinalized()).isTrue();
+        assertThat(savedFlowLog.getStateStatus()).isEqualTo(StateStatus.SUCCESSFUL);
+        assertThat(savedFlowLog.getOperationType()).isEqualTo(OperationType.DIAGNOSTICS);
+        assertThat(savedFlowLog.getCloudbreakNodeId()).isEqualTo(NODE_ID);
+        assertThat(savedFlowLog.getVariables()).isEqualTo(JsonWriter.objectToJson(params));
+        assertThat(savedFlowLog.getVariablesJackson()).isEqualTo(TypedJsonUtil.writeValueAsStringSilent(params));
+    }
+
+    private void prepareFinalization() throws TransactionExecutionException {
         when(transactionService.required(any(Supplier.class))).thenAnswer(invocation -> ((Supplier) invocation.getArguments()[0]).get());
-        FlowLog lastFlowLog = new FlowLog(resourceId, FLOW_ID, "currentState", false, StateStatus.SUCCESSFUL, OperationType.UNKNOWN);
-        lastFlowLog.setId(flowDatabaseId);
+        FlowLog lastFlowLog = new FlowLog(ID, FLOW_ID, "currentState", false, StateStatus.SUCCESSFUL, OperationType.DIAGNOSTICS);
+        lastFlowLog.setId(DATABASE_ID);
         when(flowLogRepository.findFirstByFlowIdOrderByCreatedDesc(FLOW_ID)).thenReturn(Optional.of(lastFlowLog));
+        when(nodeConfig.getId()).thenReturn(NODE_ID);
+    }
 
-        FlowLog flowLog = underTest.terminate(resourceId, FLOW_ID);
-
+    private void verifyFinalization() {
         verify(flowLogRepository).finalizeByFlowId(FLOW_ID);
-        verify(flowLogRepository).updateLastLogStatusInFlow(flowDatabaseId, StateStatus.SUCCESSFUL);
-        verify(flowLogRepository).save(any());
-        verify(applicationFlowInformation).handleFlowFail(flowLog);
+        verify(flowLogRepository).updateLastLogStatusInFlow(DATABASE_ID, StateStatus.SUCCESSFUL);
+        verify(flowLogRepository).save(savedFlowLogCaptor.capture());
+    }
+
+    @Test
+    void testSaveFlowChain() {
+
     }
```

Comment:
Was this supposed to be a test for `saveChain()`? :smile:

Reply:
true, removed